tutorials/harfbuzz/src/main.rs

```diff
@@ -0,0 +1,17 @@
+fn main() {
+    use harfbuzz::{sys, Buffer, Direction, UnicodeFuncsBuilder};
+    use icu_harfbuzz::AllUnicodeFuncs;
+    let mut b = Buffer::with("مساء الخير");
+    let mut builder = UnicodeFuncsBuilder::new_with_empty_parent().unwrap();
+    //  Note: AllUnicodeFuncs is zero-sized, so these boxes don't allocate memory.
+    builder.set_general_category_func(Box::new(AllUnicodeFuncs));
+    builder.set_combining_class_func(Box::new(AllUnicodeFuncs));
+    builder.set_mirroring_func(Box::new(AllUnicodeFuncs));
+    builder.set_script_func(Box::new(AllUnicodeFuncs));
+    builder.set_compose_func(Box::new(AllUnicodeFuncs));
+    builder.set_decompose_func(Box::new(AllUnicodeFuncs));
+    b.set_unicode_funcs(&builder.build());
```

Comment:
For compiled data we could move the creation of the `UnicodeFuncs` into `icu_harfbuzz`:

```suggestion
    b.set_unicode_funcs(&AllUnicodeFuncs::new());
```

I would like to consider the comment
```rust
    //  Note: AllUnicodeFuncs is zero-sized, so these boxes don't allocate memory.
```
an implementation detail.

Reply:
UnicodeFuncsBuilder is in harfbuzz and it directly invokes the sys functions, so it can't go into the icu_harfbuzz crate, except perhaps as macro_rules.

Reply:
I think the zero-sized type qualifier is important since they need to make 6 of these boxes but that aren't actually allocating anything in the process.

Reply:
I think it's so important that it should be in `icu_harfbuzz`, not just a tutorial.

```rust
#[cfg(feature = "compiled_data")]
impl AllUnicodeFuncs {
    /// Construct a new boxed [`AllUnicodeFuncs`]. As [`AllUnicodeFuncs`] is zero-sized,
    /// this does not allocate memory
    pub fn boxed() -> Box<Self> {
        Box::new(Self)
    }
}

Reply:
I'm fine with `AllUnicodeFuncs::boxed()`; was thinking of suggesting that but wasn't convinced it was that much better than `Box::new(AllUnicodeFuncs)`. However, it does give us a nice place to document the invariant.